### PR TITLE
fix(): regression of `getPointer` usages + BREAKING: drop event data

### PR DESCRIPTION
**Pull request by ShaMan123**
## Motivation

prepare for #9175 

fixes bug from https://github.com/fabricjs/fabric.js/commit/b6f076cc0c817cd7f70abc7028c9d04295d121a8

## Description

Somewehre in v6 `_pointer` and `_abolutePointer` were changed to `getPointer` and many ~done WRONG~ were swapped in values.
Not a single test catch it so I have added tests

`pointer` => `getPointer(e, true)`
`absolutePointer`  => `getPointer(e)`

### CODE REVIEW

**Review by ShaMan123 — COMMENTED** [review]
step

**Comment by asturur** [comment]
@ShaMan123 those are the only swap instances i could find. Do you think there are more?

I removed the identity matrix test and i added a more generic matrix that doesn't put the absolute pointer in 0,0

File: src/canvas/Canvas.ts [hunk 1]
```diff
@@ -512,6 +512,8 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
       target,
       subTargets: targets,
       dragSource: this._dragSource,
+      // this is an absolute pointer, the name convention is wrong.
+      // such was in 5.x
```
Comment (asturur) [inline src/canvas/Canvas.ts:516]: What do we want to do with this? do we make consistency and we fix it?

**Review by asturur — DISMISSED** [review]
let with my proposal of normalization. If this is good for you is approved you can merge.

File: src/canvas/Canvas.ts (hunk 1 above)
Comment (asturur) [reply src/canvas/Canvas.ts]: update: i normalized it. Do not merge if you don't agree.

**Review by asturur — DISMISSED** [review]

**Review by asturur — APPROVED** [review]

File: src/canvas/Canvas.ts (hunk 1 above)
Comment (ShaMan123) [reply src/canvas/Canvas.ts]: Yes
I think we should expose a protected method that build the event data so everything calls it

File: src/canvas/Canvas.ts [hunk 2]
```diff
@@ -513,8 +513,8 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
       subTargets: targets,
       dragSource: this._dragSource,
       // this is an absolute pointer, the name convention is wrong.
```
Comment (ShaMan123) [inline src/canvas/Canvas.ts:515]:
```suggestion
```

**Review by ShaMan123 — COMMENTED** [review]
Brush API also uses the wrong naming... but that was like that before v6 => to be done in  #8476 IMO
Group selector also uses the absolute pointer, which is weird cause it is not affected by vpt

getPointer usages seem correct

**Comment by ShaMan123** [comment]
you anyways need to approve it since I created it, so merge at will

**Review by ShaMan123 — COMMENTED** [review]
done?

**Comment by asturur** [comment]
I can't find any getPointer usage in the brush files, can you point to me?
with this PR any usage of getPointer function is associated to the correct name, or at least that seems  to me

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
At least the pointers in brush are exposed to the brush functionality only, that is an issue for who is creating custom brushes that should be way less than who is just listening to events

**Pull request merged**